src/main/java/com/aws/greengrass/util/platforms/unix/linux/LinuxSystemResourceController.java

```diff
@@ -63,30 +63,36 @@ public void updateResourceLimits(GreengrassService component, Map<String, Object
             }
             if (resourceLimit.containsKey(MEMORY_KEY)) {
                 long memoryLimitInKB = Coerce.toLong(resourceLimit.get(MEMORY_KEY));
-
-                // TODO: add input validation
-                String memoryLimit = Long.toString(memoryLimitInKB * 1024);
-                Files.write(Cgroup.Memory.getComponentMemoryLimitPath(component.getServiceName()),
-                        memoryLimit.getBytes(StandardCharsets.UTF_8));
+                if (memoryLimitInKB > 0) {
+                    String memoryLimit = Long.toString(memoryLimitInKB * 1024);
+                    Files.write(Cgroup.Memory.getComponentMemoryLimitPath(component.getServiceName()),
+                            memoryLimit.getBytes(StandardCharsets.UTF_8));
+                } else {
+                    logger.atWarn().kv(COMPONENT_NAME, component.getServiceName()).kv(MEMORY_KEY, memoryLimitInKB)
+                            .log("The provided memory limit is invalid");
+                }
             }
 
             if (!Files.exists(Cgroup.CPU.getSubsystemComponentPath(component.getServiceName()))) {
                 initializeCgroup(component, Cgroup.CPU);
             }
             if (resourceLimit.containsKey(CPU_KEY)) {
                 double cpu = Coerce.toDouble(resourceLimit.get(CPU_KEY));
+                if (cpu > 0) {
+                    byte[] content = Files.readAllBytes(
+                            Cgroup.CPU.getComponentCpuPeriodPath(component.getServiceName()));
+                    int cpuPeriodUs = Integer.parseInt(new String(content, StandardCharsets.UTF_8).trim());
```

Comment:
nit: also use `Coerce` for parsing?

Reply:
Not sure about that. This has nothing to do with our config Topics. I would rather keep it simple.